res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -158,147 +163,123 @@ VCI102.pitch = function(ch, midino, value, status, group) {
     }
+    engine.setValue(group, key, value < 127 ? value / 128 : 1);
 };
 
-VCI102.loopLength = [4, 4, 4, 4];
-
-VCI102.setLoopLength = function(ch, status, value) {
-    var LED = [
-        [0x33, 0x29],
-        [0x29, 0x26],
-        [0x33, 0x29],
-        [0x29, 0x26]
-    ];
-    if (value > 64 || value * 32 < 1) return;
-    VCI102.loopLength[ch] = value;
-    midi.sendShortMsg(status, LED[ch][0], (value < 4) * 127);
-    midi.sendShortMsg(status, LED[ch][1], (value > 4 || value * 4 < 1) * 127);
+VCI102.parameter1 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter1");
+    } else {
+        VCI102.meta(ch, value, VCI102.slot(group, 1), "meta");
+    }
 };
 
-VCI102.loop = function(ch, midino, value, status, group) {
-    if (value) {
-        if (engine.getValue(group, "loop_enabled")) {
-            engine.setValue(group, "reloop_exit", 1);
-        } else {
-            engine.setValue(group, "beatloop", VCI102.loopLength[ch]);
-        }
+VCI102.parameter2 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter2");
+    } else {
+        VCI102.meta(ch, value, VCI102.slot(group, 2), "meta");
     }
 };
 
-VCI102.reloop = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_out", value > 0);
+VCI102.parameter3 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter3");
     } else {
-        engine.setValue(group, "reloop_exit", 1);
+        VCI102.meta(ch, value, VCI102.slot(group, 3), "meta");
     }
 };
 
-VCI102.loop_halve = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_halve", value > 0);
-    } else if (value) {
-        VCI102.setLoopLength(ch, status, VCI102.loopLength[ch] / 2);
+VCI102.parameter4 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter4");
+    } else {
+        VCI102.meta(ch, value, group, "super1");
     }
 };
 
-VCI102.loop_double = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_double", value > 0);
-    } else if (value) {
-        VCI102.setLoopLength(ch, status, VCI102.loopLength[ch] * 2);
+VCI102.prev_effect = function(ch, midino, value, status, group) {
+    var key, n = engine.getValue(group, "focused_effect");
+    if (n) {
+        group = VCI102.slot(group, n);
+        key = "prev_effect";
```

Comment:
prev/next_chain aren't very useful at the moment. In the future (after Mixxx 2.1) I think they'll be revived to cycle through user defined chain presets.